### ISSUE TITLE
Issue with building flutter app on IOS due to not able to install AppAuth 1.4.0

### ISSUE THREAD

**Issue by raykleijngeld** [flutter/flutter#83470]
Hi, 
After I upgraded all my library dependencies using flutter upgrade --major-version I am no longer able to build the app for iOS.

During the build phase and pod install it tries to download AppAuth 1.4.0 and I get an error in the clone statement.
I have tried to run the statement manually and I am getting the same error.

```console
The error is:
[!] Error installing AppAuth
[!] /usr/local/bin/git clone https://github.com/openid/AppAuth-iOS.git /var/folders/0d/z_y7jpq177gbzb76f_yrpgy80000gn/T/d20210517-11798-xhkn8t --template= --single-branch --depth 1 --branch 1.4.0

Cloning into '/var/folders/0d/z_y7jpq177gbzb76f_yrpgy80000gn/T/d20210517-11798-xhkn8t'...
warning: Could not find remote branch 1.4.0 to clone.
fatal: Remote branch 1.4.0 not found in upstream origin
```

Please note: android version builds and works without any issues.

I have deleted the podfile and did a full clean.

I believe the AppAuth is related to the flutter_google_sign_in plugin:
 `google_sign_in: ^4.4.4`

Any help would be really appreciated.

Thanks in advance,

Ray

**Comment by TahaTesser** [comment]
Hi @raykleijngeld 
Please try the latest `google_sign_in: ^5.0.4` 

If the problem persists, can you please provide your `flutter doctor -v` with the upgrade, complete reproducible minimal code sample, and complete `flutter run --verbose` logs running the minimal code sample?
Thank you

**Comment by raykleijngeld** [comment]
Hi Taha,
Thanks for the quick response.
I have changed my google_sign_in package dependency to 5.0.4 as suggested but I am still having the same issue (see below).

  # * Auth
    google_sign_in: ^5.0.4
    local_auth: ^1.1.5
    # local_auth: ^0.6.1
    permission_handler: ^7.1.0

To create a reproducible code snippet I have to strip everything out of our application which will take some time.

Meanwhile I hope the flutter doctor --verbose output will give a bit more information as to why this happens.

Thanks
Ray


Analyzing dependencies
cloud_firestore: Using Firebase SDK version '7.11.0' defined in 'firebase_core'
firebase_core: Using Firebase SDK version '7.11.0' defined in 'firebase_core'
firebase_messaging: Using Firebase SDK version '7.11.0' defined in 'firebase_core'
Downloading dependencies
Installing AppAuth (1.4.0)

[!] Error installing AppAuth
[!] /usr/local/bin/git clone https://github.com/openid/AppAuth-iOS.git /var/folders/0d/z_y7jpq177gbzb76f_yrpgy80000gn/T/d20210531-2238-1eng2fj --template= --single-branch --depth 1 --branch 1.4.0

Cloning into '/var/folders/0d/z_y7jpq177gbzb76f_yrpgy80000gn/T/d20210531-2238-1eng2fj'...
warning: Could not find remote branch 1.4.0 to clone.

I have added my flutter doctor -v output below:
[✓] Flutter (Channel stable, 2.0.6, on macOS 11.3.1 20E241 darwin-x64, locale en-AU)
    • Flutter version 2.0.6 at /Users/ray/Downloads/flutter
    • Framework revision 1d9032c7e1 (4 weeks ago), 2021-04-29 17:37:58 -0700
    • Engine revision 05e680e202
    • Dart version 2.12.3

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at /Users/ray/Library/Android/sdk
    • Platform android-30, build-tools 30.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✗] Chrome - develop for the web (Cannot find Chrome executable at /Applications/Google
    Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] VS Code (version 1.47.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.1

[!] Connected device
    ! No devices available

! Doctor found issues in 3 categories.

**Comment by raykleijngeld** [comment]
Steps to reproduce: 
Create a new flutter app: flutter create demo_app
When building without any dependencies no issues.
When I add google_sign_in: ^5.0.4 I get the error as mentioned before.

**Comment by raykleijngeld** [comment]
When I change the google_sign_in version back to ^3.0.0 I don't get an error on AppAuth but I now get an error on GTMSessionFetcher:

Note that I can't go back to 3.0.0 because of other dependencies.

ray@Rays-iMac ios % pod install
Analyzing dependencies
Downloading dependencies
Installing Flutter (1.0.0)
Installing GTMSessionFetcher (1.5.0)

[!] Error installing GTMSessionFetcher
[!] /usr/local/bin/git clone https://github.com/google/gtm-session-fetcher.git /var/folders/0d/z_y7jpq177gbzb76f_yrpgy80000gn/T/d20210531-7449-mjpklu --template= --single-branch --depth 1 --branch v1.5.0

Cloning into '/var/folders/0d/z_y7jpq177gbzb76f_yrpgy80000gn/T/d20210531-7449-mjpklu'...
warning: Could not find remote branch v1.5.0 to clone.
fatal: Remote branch v1.5.0 not found in upstream origin

**Comment by raykleijngeld** [comment]
Hmm I think it is an issue with MacOS git.. 
When I run the statement: git clone https://github.com/google/gtm-session-fetcher.git --template= --single-branch --depth=1 --branch v1.5.0 on my window system it runs without any issues. Windows git version: 2.21.0.windows.1 

However when I run the same statement on my macOS I get the error:

git clone https://github.com/google/gtm-session-fetcher.git --template= --single-branch --depth=1 --branch v1.5.0
Cloning into 'gtm-session-fetcher'...
warning: Could not find remote branch v1.5.0 to clone.
fatal: Remote branch v1.5.0 not found in upstream origin
![gitwindows](https://user-images.githubusercontent.com/84309072/120135379-7ea76980-c213-11eb-9315-bb6cb10619c8.png)

Git version MacOs:git version 2.28.0

**Comment by raykleijngeld** [comment]
It was indeed related to git. But not the git version but it turned out that in my global git config setting remote.origin.url was set which caused the git clone command to fail.

I worked this out by trying to reinstall homebrew and noticed our flutter app was downloaded as well.

After unsetting the remote.origin.url from global config I was able to build the app succesfully.

git config --global --unset remote.origin.url
